src/main/java/net/dv8tion/jda/webhook/WebhookClient.java

```diff
@@ -406,52 +366,51 @@ protected void backoffQueue()
 
     protected void drainQueue()
     {
-        while (!queue.isEmpty())
-        {
-            final Pair<RequestBody, CompletableFuture<?>> pair = queue.peek();
+        while (!queue.isEmpty()) {
+            final Pair<RequestBody, CompletableFuture<SendWebhookMessage>> pair = queue.peek();
             executePair(pair);
         }
         isQueued = false;
     }
 
-    private boolean enqueuePair(@Async.Schedule Pair<RequestBody, CompletableFuture<?>> pair)
+    private boolean enqueuePair(@Async.Schedule Pair<RequestBody, CompletableFuture<SendWebhookMessage>> pair)
     {
         return queue.add(pair);
     }
 
-    private void executePair(@Async.Execute Pair<RequestBody, CompletableFuture<?>> pair)
+    private void executePair(@Async.Execute Pair<RequestBody, CompletableFuture<SendWebhookMessage>> pair)
     {
-        if (pair.getRight().isCancelled())
-        {
+        if (pair.getRight().isCancelled()) {
             queue.poll();
             return;
         }
 
         final Request request = newRequest(pair.getLeft());
-        try (Response response = client.newCall(request).execute())
-        {
+        try (Response response = client.newCall(request).execute()) {
             bucket.update(response);
-            if (response.code() == Bucket.RATE_LIMIT_CODE)
-            {
+            if (response.code() == Bucket.RATE_LIMIT_CODE) {
                 backoffQueue();
                 return;
-            }
-            else if (!response.isSuccessful())
-            {
+            } else if (!response.isSuccessful()) {
                 final HttpException exception = failure(response);
                 LOG.error("Sending a webhook message failed with non-OK http response", exception);
                 queue.poll().getRight().completeExceptionally(exception);
                 return;
             }
-            queue.poll().getRight().complete(null);
-            if (bucket.isRateLimit())
-            {
+
+            SendWebhookMessage returnedMessage = null;
+
+            if (isWait) {
+                JSONObject messageJson = new JSONObject(new String(IOUtil.readFully(Requester.getBody(response))));
+                returnedMessage = new SendWebhookMessageFactory().create(messageJson);
```

Comment:
What is the point of creating this factory as an object if it does not have any state?